test/exlasticsearch/repo/mapping_types_test.exs

```
@@ -0,0 +1,325 @@
+defmodule ExlasticSearch.Repo.MappingTypesTest do
+  use ExUnit.Case, async: true
+
+  alias ExlasticSearch.Aggregation
+  alias ExlasticSearch.MultiVersionTestModel, as: MVTestModel
+  alias ExlasticSearch.Query
+  alias ExlasticSearch.Repo
+  alias ExlasticSearch.TestModel
+  alias ExlasticSearch.TestModel2
+
+  require Logger
+
+  setup_all do
+    Repo.delete_index(TestModel)
+    Repo.create_index(TestModel)
+    {:ok, %{status_code: 200}} = Repo.create_mapping(TestModel, :index, include_type_name: true)
```

the test setups weren't matching against anything for the return value of `Repo.create_mapping(...)`, but that also covered up that they were silently failing on creating mappings for deprecated mapping types, so I put in matching against `{:ok, ...}` _and_ that the status code is 200, so if these fail to create we'd at least know about it.